src/consumer/stream_consumer.rs

```diff
@@ -175,43 +177,25 @@ impl<C: ConsumerContext> StreamConsumer<C> {
     pub fn start_with(&self, poll_interval: Duration, no_message_error: bool) -> MessageStream<C> {
         // TODO: verify called once
         let (sender, receiver) = mpsc::channel(CONSUMER_CHANNEL_SIZE);
-        let consumer = self.consumer.clone();
-        let should_stop = self.should_stop.clone();
-        let handle = thread::Builder::new()
-            .name("poll".to_string())
-            .spawn(move || {
-                poll_loop(
-                    consumer.as_ref(),
-                    sender,
-                    should_stop.as_ref(),
-                    poll_interval,
-                    no_message_error,
-                );
-            })
-            .expect("Failed to start polling thread");
-        *self.handle.lock().unwrap() = Some(handle);
+        tokio::spawn(poll_loop(
```

Comment:
this will keep polling even if StreamConsumer is dropped, right?

Reply:
No because on drop `should_stop` is set to `true` and `poll_loop` will stop looping and return.

Reply:
Yep, exactly. Actually waiting for the task to drop requires support for an `AsyncDrop` sort of thing, which doesn't exist at the moment. It's not necessary to wait, though, because the `Arc<BaseConsumer>` will keep the consumer alive as long as the task is alive.